autoblocks/_impl/global_state.py

```diff
@@ -25,10 +29,20 @@ def init() -> None:
         daemon=True,
     )
     background_thread.start()
-
+    signal.signal(signal.SIGTERM, _on_exit)
+    signal.signal(signal.SIGINT, _on_exit)
     _started = True
 
 
+def _on_exit(signum, frame):
+    if _loop and _loop.is_running():
+        log.info(f"Attempting to gracefully stop event loop. Received signal {signum}")
+        _loop.stop()
+        pending = asyncio.all_tasks()
+        _loop.run_until_complete(asyncio.gather(*pending))
+    exit(0)
```

Comment:
can we avoid setting the exit code ourselves?

Reply:
I actually think we can just straight omit it, since this would be the last block of code to run before exit

Reply:
i think we need this actually, when testing locally it never exits